### PR TITLE
Add rejection reason dropdown

### LINKED ISSUE #286: Add reason for rejection
Optionally have a reason for rejecting slides:
image

### PR DESCRIPTION
**Pull request by naglepuff**
Resolves #286 

Now, deployments of WSI DeID can be configured to require a reason for rejecting an image from one of the WSI DeID workflow folder. There are two new settings in the girder configuration

1. `require_reject_reason`: set to `True` to require users to select a reason for rejecting an image. Setting this to `False` will allow rejection with no reason given.
2. `reject_reasons`: This is a list of objects that describe the possible reasons for rejection.

Users can select a reason from the drop down next to the `Reject` buttons from the item page.

This information is stored as item metadata under the key `reject`.

To do:

- [x] Update readmes where appropriate

### CODE REVIEW

File: wsi_deid/web_client/templates/ItemView.pug [hunk 1]
```diff
@@ -29,11 +29,24 @@
       select.g-refile-select
         option(value='__none__') Manually specify
       span(title="Refile with this TokenID (folder) name")
-        | TokenID
+        | TokenIg-workflow-buttonD
```
Comment (manthey) [inline wsi_deid/web_client/templates/ItemView.pug:32]: I think this was unintended.

Comment (naglepuff) [reply wsi_deid/web_client/templates/ItemView.pug]: Whoops. This has been fixed in 4453624

File: devops/wsi_deid/girder.local.conf [hunk 2]
```diff
@@ -59,3 +60,24 @@ folder_name_field = "TokenID"
 image_name_field = ""
 validate_image_id_field = False
 new_token_pattern = "####@@######"
+reject_reasons = [{
```
Comment (manthey) [inline devops/wsi_deid/girder.local.conf:63]: We should probably add a default to this value in the code (so if it isn't set in the config AND `require_reject_reason = True`, you can still redact the image.  Either that or if it isn't set, ignore the value of `require_reject_reason`.

File: wsi_deid/web_client/templates/ItemView.pug [hunk 3]
```diff
@@ -36,4 +36,17 @@
         input.g-refile-imageid
       button.g-refile-button.btn(class='btn-'+buttons.refile.color, action=button, title=buttons.refile.title) #{buttons.refile.name}
   for button in buttonList
-    button.g-workflow-button.btn(class='btn-'+buttons[button].color, action=button, title=buttons[button].title) #{buttons[button].name}
+    - var isDisabled = button === 'reject' && require_reject_reason
+    button.g-workflow-button.btn(class='btn-'+buttons[button].color, action=button, title=buttons[button].title, disabled=isDisabled) #{buttons[button].name}
+  if buttonList.includes('reject') && rejection_reasons.length > 0
```
Comment (manthey) [inline wsi_deid/web_client/templates/ItemView.pug:41]: This should also check if the reject reason is required (and, if not, not show the drop down).

Comment (naglepuff) [reply wsi_deid/web_client/templates/ItemView.pug]: Added a check in 5ade89f

File: devops/wsi_deid/girder.local.conf (hunk 2 above)
Comment (naglepuff) [reply devops/wsi_deid/girder.local.conf:63]: Added defaults based on the issue in bd5d4f9

**Comment by naglepuff** [comment]
> We will want to include the rejection reason in the export report. Maye a column at the end `RejectionReason` that would be populated when appropriate? That could be a follow-up PR or in this one.

I took a crack at this in b899f16. I added the column closer to the Status column, thinking that might be easier to read, but if rejection isn't a common workflow I can definitely see moving it to the end, since an extra blank column is not useful. It's a very quick change so please let me know if you disagree with my choice.

**Comment by naglepuff** [comment]
After doing investigation for issue #319 I think the following should be done here:

- [ ] Undo all changes to `girder.local.conf`
- [ ] Add new settings and defaults to `config.py`'s `defaultConfig` dict
- [ ] Remove `RejectReasons` export from `utils.js` and use the server settings instead

**Comment by naglepuff** [comment]
> We have other defaults that are in config.py in the girder.local.conf, so I think leaving it there is more consistent than removing it (otherwise we need to show an example girder conf with it present).

I've moved the default settings from `utils.js` to `config.py` in 63de9e6 and c74f693

**Review by manthey — APPROVED** [review]

**Pull request merged**